src/sonic-config-engine/sonic_device_util.py

```diff
@@ -73,10 +108,14 @@ def get_system_mac():
         # Try valid mac in eeprom, else fetch it from eth0
         platform = get_platform_info(get_machine_info())
         hwsku = get_machine_info()['onie_machine']
-        profile_cmd = 'cat /usr/share/sonic/device/' + platform +'/'+ hwsku +'/profile.ini | cut -f2 -d='
+        profile_cmd = 'cat' + SONIC_DEVICE_PATH + '/' + platform +'/'+ hwsku +'/profile.ini | cut -f2 -d='
         hw_mac_entry_cmds = [ profile_cmd, "sudo decode-syseeprom -m", "ip link show eth0 | grep ether | awk '{print $2}'" ]
     else:
-        hw_mac_entry_cmds = [ "ip link show eth0 | grep ether | awk '{print $2}'" ]
+        mac_address_cmd = "cat /sys/class/net/eth0/address"
+        if namespace is not None:
+            mac_address_cmd = "sudo ip netns exec {} {}".format(namespace, mac_address_cmd)
```

Comment:
what is the container eth0 mac address? how can make sure it is unique? is it randomly generated?

Reply:
The container eth0 mac is a dummy mac. 
In the current design all the frontend asics will use the eth0 mac whereas the backend asics will use the container's  eth0 mac. 

Corresponding change in sonic-cfggen
```
if asic_name is not None and asic_role.lower() == "backend":
            mac = get_system_mac(asic_name)
        else:
            mac = get_system_mac()
```